internal/services/containers/kubernetes_fleet_update_strategy_resource_test.go

```diff
@@ -0,0 +1,183 @@
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_fleet_update_strategy", "test")
+	r := KubernetesFleetUpdateStrategyTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccKubernetesFleetUpdateStrategy_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_fleet_update_strategy", "test")
+	r := KubernetesFleetUpdateStrategyTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+func (r KubernetesFleetUpdateStrategyTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := fleetupdatestrategies.ParseUpdateStrategyID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Containers.FleetUpdateStrategiesClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+func (r KubernetesFleetUpdateStrategyTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kubernetes_fleet_update_strategy" "test" {
+  name             = "acctestfus-%[2]d"
+  fleet_manager_id = azurerm_kubernetes_fleet_manager.test.id
+  strategy {
+    stage {
+      name = "acctestfus-%[2]d"
+      group {
+        name = "acctestfus-%[2]d"
+      }
+    }
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r KubernetesFleetUpdateStrategyTestResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kubernetes_fleet_update_strategy" "import" {
+  name             = azurerm_kubernetes_fleet_update_strategy.test.name
+  fleet_manager_id = azurerm_kubernetes_fleet_update_strategy.test.fleet_manager_id
+  strategy {
+    stage {
+      name = "acctestfus-%[2]d"
+      group {
+        name = "acctestfus-%[2]d"
+      }
+    }
+  }
+}
+`, r.basic(data), data.RandomInteger)
+}
+
+func (r KubernetesFleetUpdateStrategyTestResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kubernetes_fleet_update_strategy" "test" {
+  name             = "acctestfus-%[2]d"
+  fleet_manager_id = azurerm_kubernetes_fleet_manager.test.id
+  strategy {
+    stage {
+      name = "acctestfus-%[2]d"
```

Comment:
Can we test by updating the stage name and group name, and observe whether the resource is not recreated?